apps/retrospective/models/retrospective.go

```diff
@@ -35,12 +37,19 @@ func (retrospective *Retrospective) Validate(db *gorm.DB) (err error) {
 		err = errors.New("story points per week cannot be negative")
 		return err
 	}
+	if _, exists := timetracker.TimeProvidersDisplayNameMap[retrospective.TimeProviderName]; !exists {
+		return errors.New("Invalid time provider name")
+	}
 	return
 }
 
 // BeforeSave ...
 func (retrospective *Retrospective) BeforeSave(db *gorm.DB) (err error) {
-	return retrospective.Validate(db)
+	err = retrospective.Validate(db)
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
We can write `return err` instead of the above 4 lines.

Reply:
done.